docs/source/nlp/language_modeling.mdx

```diff
@@ -0,0 +1,379 @@
+ 'selftext_urls.url': ['http://apod.nasa.gov/apod/image/1201/freeflyer_nasa_3000.jpg'],
+ 'subreddit': 'askscience',
+ 'title': 'Few questions about this space walk photograph.',
+ 'title_urls.url': []}
+```
+
+Each subfield is now a separate column, as indicated by the `answers` prefix. There is another functionality you need to include in your preprocessing function. The `answers.text` field should be a string instead of a list. Create a preprocessing function to convert the list to a string and add a truncation strategy:
+
+```py
+>>> def preprocess_function(examples):
+...     return tokenizer([" ".join(x) for x in examples["answers.text"]], truncation=True)
+```
+
+Use 🤗 Datasets [`map`](https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.Dataset.map) function to apply the preprocessing function over the entire dataset. You can speed up the preprocessing by setting `batched=True` and increasing the number of processes with `num_proc`. Remove the columns you don't need:
+
+```py
+>>> tokenized_eli5 = eli5.map(
+...     preprocess_function,
+...     batched=True,
+...     num_proc=4,
+...     remove_columns=[
+...         "q_id",
+...         "title",
+...         "selftext",
+...         "document",
+...         "subreddit",
+...         "answers.a_id",
+...         "answers.text",
+...         "answers.score",
+...         "title_urls.url",
+...         "selftext_urls.url",
+...         "answers_urls.url",
+...     ],
+... )
+```
+
+Now you need a second preprocessing function to:
+
+- Concatenate all the text.
+- Split the concatenated text into smaller chunks defined by `block_size`.
+- Create a copy of the inputs for the labels because the model shifts the labels to the right.
+
+```py
+>>> block_size = 128
+
+
+>>> def group_texts(examples):
+...     concatenated_examples = {k: sum(examples[k], []) for k in examples.keys()}
+...     total_length = len(concatenated_examples[list(examples.keys())[0]])
+...     total_length = (total_length // block_size) * block_size
+...     result = {
+...         k: [t[i : i + block_size] for i in range(0, total_length, block_size)]
+...         for k, t in concatenated_examples.items()
+...     }
+...     result["labels"] = result["input_ids"].copy()
+...     return result
+```
+
+Apply the `group_texts` function over the entire dataset, and speed up the operation by batching examples and increasing the number of processes:
+
+```py
+>>> lm_dataset = tokenized_eli5.map(
+...     group_texts,
+...     batched=True,
+...     num_proc=4,
+... )
+```
+
+## Causal language modeling
+
+CLM is often used for text generation. This section shows you how to fine-tune a causal language model to generate new text.
```

Comment:
I'd avoid using acronyms in written doc. Writing Causal language modeling is just as easy to read, and clearer to the user.